src/utils.cpp

```diff
@@ -934,6 +935,16 @@ bool utils::is_valid_podcast_type(const std::string& mimetype)
 	return rs_is_valid_podcast_type(mimetype.c_str());
 }
 
+LinkType utils::podcast_mime_to_link_type(const std::string& mimetype, bool& ok)
+{
+	const auto result = rs_podcast_mime_to_link_type(mimetype.c_str(), &ok);
+	if (ok) {
+		return static_cast<LinkType>(result);
+	}
+
+	return LinkType::AUDIO;
```

Comment:
I would be fine with pulling in optional-lite just to get rid of this (given that we move to c++17 at some point).

Reply:
I guess I'll hold off until we have more uses for `std::optional`.

Main points against header-only dep is: 1) it increases compilation times; 2) maintainers downstream have to jump through hoops in order to replace it with a version provided by the distro (and they might have to package it first). IMHO a bit of inconvenience doesn't outweigh those points (yet).